sdks/java/nexmark/src/main/java/org/apache/beam/sdk/nexmark/Main.java

```diff
@@ -74,22 +96,89 @@ void runAll(OptionT options, NexmarkLauncher nexmarkLauncher) throws IOException
           appendPerf(options.getPerfFilename(), configuration, perf);
           actual.put(configuration, perf);
           // Summarize what we've run so far.
-          saveSummary(null, configurations, actual, baseline, start);
+          saveSummary(null, configurations, actual, baseline, start, options);
         }
       }
+      if (options.getExportSummaryToBigQuery()){
+        savePerfsToBigQuery(options, actual, null);
+      }
     } finally {
       if (options.getMonitorJobs()) {
         // Report overall performance.
-        saveSummary(options.getSummaryFilename(), configurations, actual, baseline, start);
+        saveSummary(options.getSummaryFilename(), configurations, actual, baseline, start, options);
         saveJavascript(options.getJavascriptFilename(), configurations, actual, baseline, start);
       }
     }
-
     if (!successful) {
       throw new RuntimeException("Execution was not successful");
     }
   }
 
+  @VisibleForTesting
+  static void savePerfsToBigQuery(
+      NexmarkOptions options,
+      Map<NexmarkConfiguration, NexmarkPerf> perfs,
+      @Nullable FakeBigQueryServices fakeBigQueryServices) {
+    Pipeline pipeline = Pipeline.create(options);
```

Comment:
I'm still not sure why we can't create a new PipelineOptions object here (PipelineOptions newOptions = PipelineOptionsFactory.create()) and use DirectRunner. That will be much faster (and cleaner) than using the original options with original runner.

Reply:
Yes it is technically feasible to create a new PipelineOptions with runner == DirectRunner and use it in the second pipeline. But where I'm not convinced is that it will require to ship direct runner libs in nexmark even when we are running the queries on another runner. It might be problematic to have them in the fat jar deployed on a spark cluster for example. 
Currently, we only ship direct runner libs in the classpath when we run JVM local tests on the direct runner (profile).
Please note that the other big query additions to nexmark (sink the output PCollection to big query) currently run with the same runner than the queries in the same pipeline)

Reply:
Also an extra point is that the volume data to insert into BQ is at maximum 12 (queries) x 3 fields for each nexmark run; thus I don't think there is any performance/speed concern here.